src/main/java/ch/njol/skript/expressions/ExprExactItem.java

```diff
@@ -0,0 +1,47 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import org.bukkit.block.Block;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.BlockStateMeta;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Exact Item")
+@Description(
+	"Get an exact item representation of a block, carrying over any data. "
+	+ "For example, using this expression on a chest block with items stored inside will return a chest "
+	+ "item with the exact same items in its inventory as the chest block."
+)
+@Examples("set {_item} to exact item of block at location(0, 0, 0)")
+@Since("INSERT VERSION")
+public class ExprExactItem extends SimplePropertyExpression<Block, ItemStack> {
+
+	static {
+		register(ExprExactItem.class, ItemStack.class, "exact item[s]", "blocks");
+	}
+
+	@Override
+	public @Nullable ItemStack convert(Block block) {
+		ItemStack itemStack = new ItemStack(block.getType());
```

Comment:
You might need to verify that the `Material` is representable as an ItemStack (e.g. `Material#isItem`)